snmp/datadog_checks/snmp/snmp.py

```diff
@@ -98,13 +106,50 @@ def _get_instance_key(self, instance):
 
         return key
 
+    def discover_instances(self):
+        network = ipaddress.ip_network(self._config.network_address)
+        discovery_interval = self._config.instance.get('discovery_interval', 3600)
+        discovery_pause = self._config.instance.get('discovery_pause', 1)
+        while self._running:
+            start_time = time.time()
+            for host in network.hosts():
+                host = str(host)
+                if host in self._config.discovered_instances:
+                    continue
+                instance = self._config.instance.copy()
+                instance.pop('network_address')
+                instance['ip_address'] = host
+
+                config = InstanceConfig(
```

Comment:
does it need tags?

Reply:
What do you mean needs? Tags are passed through `instance`.